Commands/SynchronizeUsers.php

```diff
@@ -92,6 +92,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
             }
         }
 
+        $localLogins = array_diff($this->usersManagerAPI->getUsersLogin(), $logins);
```

Comment:
This will change the default behavior of the command which may surprise some users that are using it in an automated workflow. Can you add an option to the command like `--remove-deleted-users` that must be specified for this new logic to be used?

Reply:
Also I think the array_diff here is backwards. We don't want to remove every user that isn't in `$logins`, that will include users that may be in LDAP but were not specified in he command. Instead, can you keep track of which logins were successfully synchronized and which were not due to the login not existing in LDAP in the above loop, and only delete those?

Reply:
This [`array_diff`](https://www.php.net/array_diff) makes a census of all Matomo users' login not coming from the current list collected from LDAP.

Reply:
This [`array_diff`](https://www.php.net/array_diff) makes a census of all Matomo users' login not coming from the current list collected from LDAP.

Reply:
Going through every matomo user means you'll be looking at logins that are not in LDAP which will result in warnings that don't need to be in the output, and it will be slower especially when there are hundreds or thousands of users. Can you fix this?

Reply:
I received an email notification for a comment I do not see anymore here. Strange?
> Going through every matomo user means you'll be looking at logins that are not in LDAP which will result in warnings that don't need to be in the output, and it will be slower especially when there are hundreds or thousands of users. Can you fix this?
The loop below will call `purgeUser` in `API`, which makes sure the user is a LDAP user before calling the actual `purgeLdapUser` function.
Hence, no warning shall ever be emitted, which would only happen is the actual purge function was to be called on an invalid (ie non-LDAP) user.